OTEP.md

```diff
@@ -371,158 +271,38 @@ sdk:
+
+### Additional method to configure OpenTelemetry
 
-* How the change would impact and interact with existing functionality
-* Likely error modes (and how to handle them)
-* Corner cases (and how to handle them)
+If the implementation suggested in this OTEP goes ahead, users will be presented with another mechanism for configuring OpenTelemetry. This may cause confusion for users who are new to the project. It may be possible to mitigate the confusion by providing users with best practices and documentation.
 
-While you do not need to prescribe a particular implementation - indeed, OTEPs should be about **behaviour**, not implementation! - it may be useful to provide at least one suggestion as to how the proposal *could* be implemented. This helps reassure reviewers that implementation is at least possible, and often helps them inspire them to think more deeply about trade-offs, alternatives, etc.
+### Many ways to configure may result in users not knowing what is configured
 
-## Trade-offs and mitigations (TBD)
+As there are multiple mechanisms for configuration, it's possible that the active configuration isn't what was expected. This could happen today, and one way it could be mitigated would be by providing a mechanism to list the active OpenTelemetry configuration.
 
-What are some (known!) drawbacks? What are some ways that they might be mitigated?
 
-Note that mitigations do not need to be complete *solutions*, and that they do not need to be accomplished directly through your proposal. A suggested mitigation may even warrant its own OTEP!
+### Errors or difficulty in configuration files
 
-## Prior art and alternatives (TBD)
+Configuration files provide an opportunity for misconfiguration. A way to mitigate this would be to provide clear messaging and fail quickly when misconfiguration occurs.
 
-What are some prior and/or alternative approaches? For instance, is there a corresponding feature in OpenTracing or OpenCensus? What are some ideas that you have rejected?
+## Prior art and alternatives
+
+The working group looked to the OpenTelemetry Collector and OpenTelemetry Operator for inspiration and guidance.
+
+### Alternative schema languages
+
+In choosing to recommend JSON schema, the working group looked at the following options:
+
+* [Cue](https://cuelang.org/) - A promising simpler language to define a schema, the working group decided against CUE because:
+  * Tooling available for validating CUE files in languages outside of Go were limited.
+  * Familiarity and learning curve would create problems for both users and contributors of OpenTelemetry.
+* [Protobuf](https://developers.google.com/protocol-buffers) - With protobuf already used heavily in OpenTelemetry, the format was worth investigating as an option to define the schema. The working group decided against Protobuf because:
+  * Validation of schema requires additional logic with custom serialization. This would need to be re-implemented in each language.
```

Comment:
I don't understand this. If using protobuf and a field is present which is part of the protobuf schema, it will automatically be validated when you try to read the config file with the schema.

Reply:
Right this was meant to capture use-cases where standard protobuf validation wouldn't be enough. I was thinking specifically of the use-case we have of naming exporters rather than providing a name field for them. Validation for this use-case works for json-schema with property patterns, but the same isn't supported in proto afaik.

Reply:
Two options for this come to mind for this:
1. Limit configuration to a single named exporter of a particular type. I.e do not support `oltp/foo: ..`, `otlp/bar: ...`. Probably ok for 99% of scenarios, but frustrating for the other 1%. Only option would be to provide "custom" user defined exporters with different names  that delegated to the OTLP exporter. I.e. implement exporters named `otlp_foo` and `otlp_bar`.
2. Treat the exporter name as an opaque key. Include an additional field to identify the type of exporter. This pattern is demonstrated [here](https://github.com/MrAlias/otel-schema/blob/main/protobuf/kitchen-sink.yaml#L98-L100):
```
exporters:
  otlp/foo:
    type: otlp
    otlp_args: ...
```

The weakness of protobuf is, IMO, that limitations in the schema definition language result in poor ergonomics if type safety is to be retained.

Reply:
I have also captured some observations wrt protobuf validation [here](https://github.com/MrAlias/otel-schema/blob/a98ada04f34ff07aeae35609d3aa801e8d8425a6/protobuf/python/README.md). In summary, maps do not support variant data type or "oneof" values, and also no way to define required fields.

Reply:
You have to use something like [AnyValue](https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/common/v1/common.proto#L28-L40) to accomplish this, which increases the verbosity. Something like this:
```
headers:
   api_key: abcd
   timeout: 1000
```
Has to become:
```
headers:
   api_key:
      string_value: abcd
   timeout:
      int_value: 1000
```